lib/temporal/workflow/errors.rb

```diff
@@ -0,0 +1,39 @@
+module Temporal
+  class Workflow
+    class Errors
+      include Concerns::Payloads
+
+      # Convert a failure returned from the server to an Error to raise to the client
+      # failure: Temporal::Api::Failure::V1::Failure
+      def error_from(failure, default_exception_class = StandardError)
+        case failure.failure_info
+        when :application_failure_info
+          exception_class = safe_constantize(failure.application_failure_info.type)
+          exception_class ||= default_exception_class
+          message = from_details_payloads(failure.application_failure_info.details)
+          backtrace = failure.stack_trace.split("\n")
+
+          exception_class.new(message).tap do |exception|
+            exception.set_backtrace(backtrace) if !backtrace.empty?
+          end
+        when :timeout_failure_info
+          TimeoutError.new("Timeout type: #{failure.timeout_failure_info.timeout_type.to_s}")
```

Comment:
We probably want `Temporal::TimeoutError` here, not Ruby's generic one

Reply:
It only does so if you expect the errors to be `require`d in the global context, which is a bad practise. This file does not `require` them, so therefore not guaranteed to resolve it